### ISSUE TITLE
Editing proc-macro crate = undefined symbol

### ISSUE THREAD

**Issue by sgrif** [rust-lang/rust#47936]
Note: This is with incremental-comp turned off, and a compiler built yesterday.

Occasionally when working on Diesel's codegen crates, I'll get an error like this:

```
error: dlsym(0x113560e00, __rustc_derive_registrar__55df2b44c5129c66de0914ff53563457_89): symbol not found
   --> diesel/src/lib.rs:121:1
    |
121 | extern crate diesel_derives2;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The only fix when this happens is to `cargo clean`. It randomly happens when I change some code, but not 100% of the time.

**Comment by nox** [comment]
I can confirm this happens.

@sgrif As a workaround to not clean *everything*, I do:

```bash
find target -name \*culprit-crate-name\* | xargs rm -rf
```

**Comment by alexcrichton** [comment]
I think this was fixed by https://github.com/rust-lang/rust/pull/47494, so closing.

**Comment by dtolnay** [comment]
Reopening because #47494 merged on Jan 18 and this was reported on a compiler built Jan 31.

**Comment by alexcrichton** [comment]
Oh oops, indeed!

**Comment by Enselic** [comment]
Triage: Is this still a problem? Wanted to ask since 5 years have passed and circumstances might have changed.
